lib/common/models/concerns/active_record_cache_aside.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Common
+  module ActiveRecordCacheAside
+    extend ActiveSupport::Concern
+
+    REDIS_CONFIG = Rails.application.config_for(:redis).freeze
+
+    included do
+      unless self < ActiveRecord::Base
+        raise ArgumentError, 'Class composing Common::ActiveRecordCacheAside must be an ActiveRecord'
+      end
+
+      def self.redis(namespace)
+        @redis_namespace = Redis::Namespace.new(namespace, redis: Redis.current)
+      end
+
+      def self.redis_ttl(ttl)
+        @redis_namespace_ttl = ttl
+      end
+
+      def self.do_cached_with(key:)
+        cached = @redis_namespace.get(key)
+        return Marshal.load(cached) if cached
```

Comment:
@kreek - Alright, so I have everything implemented, and working.  I did a bunch of test runs in development, checking the Redis CLI for confirmation.  All is good.  Nice work!!

I only made a couple minor adjustments.  One being the name of a method in `Account`, and the other being the manner in which we are serializing the object that is getting cached.  More specifically, you'll note that I favored going with the `Marshal.dump/load` approach over the `.to_json/JSON.parse` one.  Here's my thinking.  

The `Marshal` implementation will always return the actual, persisted db (Account) record from the cache.  The JSON approach would also return an instance of the (Account) class; however, it will not be the persisted one.  The non-persisted instance also has the `id` property assigned, which could be misleading.

So if in the future, should a dev get a handle on the object that is returned from the cache and want to update/delete it, as long as it is persisted, they will able to do so.  

I still have some Yard docs, and specs, to add.  Outside of that, any other thoughts?

Reply:
@hpjaj no other thoughts, I like the changes you added, this looks great! thanks!